src/Components/Replication/ReplicationSettingsForm.tsx

```diff
@@ -411,94 +506,123 @@ export default class ReplicationSettingsFormReplication extends React.Component<
           <Form.Item
             label={
               <span>
-                Filter Records&nbsp;
-                <Tooltip title="Set the labels a record should include/exclude to be replicated. If empty, all records are replicated.">
+                When Condition&nbsp;
```

Comment:
please re-use the code mirror component:

![image](https://github.com/user-attachments/assets/61bd5ace-a093-4d8c-8802-6b423380a171)